### PR TITLE
[feat] ConfigClient loadFromDataBase + small refactor

### PR DESCRIPTION
**Pull request by joshuajeschek**

### CODE REVIEW

File: src/main/java/github/weichware10/util/config/ConfigLoader.java [hunk 1]
```diff
@@ -4,6 +4,8 @@
 import com.fasterxml.jackson.databind.DatabindException;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import github.weichware10.util.Logger;
+import github.weichware10.util.db.DataBaseClient;
+
 import java.io.File;
```
Comment (weichbot10) [inline src/main/java/github/weichware10/util/config/ConfigLoader.java:9]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.CustomImportOrderCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Extra separation in import group before 'java.io.File'

File: src/main/java/github/weichware10/util/config/ConfigWriter.java [hunk 2]
```diff
@@ -4,6 +4,8 @@
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.databind.ObjectWriter;
 import github.weichware10.util.Logger;
+import github.weichware10.util.db.DataBaseClient;
+
 import java.io.BufferedWriter;
```
Comment (weichbot10) [inline src/main/java/github/weichware10/util/config/ConfigWriter.java:9]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.CustomImportOrderCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Extra separation in import group before 'java.io.BufferedWriter'

File: src/test/java/github/weichware10/util/config/ConfigurationTest.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package github.weichware10.util.config;
+
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Test;
+
+/**
```
Comment (weichbot10) [inline src/test/java/github/weichware10/util/config/ConfigurationTest.java:7]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.javadoc.SummaryJavadocCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>First sentence of Javadoc is missing an ending period.

File: src/test/java/github/weichware10/util/config/ConfigLoaderTest.java [hunk 4]
```diff
@@ -4,9 +4,15 @@
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNull;
+import static org.junit.Assert.fail;
+
+import java.util.List;
 
 import org.junit.Test;
```
Comment (weichbot10) [inline src/test/java/github/weichware10/util/config/ConfigLoaderTest.java:11]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.CustomImportOrderCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Extra separation in import group before 'org.junit.Test'

File: src/test/java/github/weichware10/util/config/ConfigLoaderTest.java [hunk 5]
```diff
@@ -4,9 +4,15 @@
...
 
+import github.weichware10.util.db.DataBaseClient;
```
Comment (weichbot10) [inline src/test/java/github/weichware10/util/config/ConfigLoaderTest.java:13]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.CustomImportOrderCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Extra separation in import group before 'github.weichware10.util.db.DataBaseClient'

Comment (weichbot10) [inline src/test/java/github/weichware10/util/config/ConfigLoaderTest.java:13]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.CustomImportOrderCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Wrong lexicographical order for 'github.weichware10.util.db.DataBaseClient' import. Should be before 'org.junit.Test'.

File: src/test/java/github/weichware10/util/config/ConfigLoaderTest.java [hunk 6]
```diff
@@ -4,9 +4,15 @@
...
+import io.github.cdimascio.dotenv.Dotenv;
```
Comment (weichbot10) [inline src/test/java/github/weichware10/util/config/ConfigLoaderTest.java:14]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.CustomImportOrderCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Wrong lexicographical order for 'io.github.cdimascio.dotenv.Dotenv' import. Should be before 'org.junit.Test'.

File: src/main/java/github/weichware10/util/config/ConfigLoader.java [hunk 7]
```diff
@@ -47,15 +48,19 @@ public static Configuration fromJson(String location) {
      * Lädt eine Konfiguration in die interne Struktur
      * ({@link Configuration}).
      *
-     * @param configId - ID der Konfiguration
+     * @param trialId - ID der Konfiguration
```
Comment (justinexner) [inline src/main/java/github/weichware10/util/config/ConfigLoader.java:51]: Eventuell "ID welche die Konfiguration enthält"

**Review by justinexner — APPROVED** [review]
Den Kommentar noch anschauen, ansonsten sehr gut gemacht 👍

File: src/main/java/github/weichware10/util/config/ConfigLoader.java (hunk 7 above)
Comment (joshuajeschek) [reply src/main/java/github/weichware10/util/config/ConfigLoader.java:51]: Nö

**Review by Rotwolf — APPROVED** [review]
Sehr gute Arbeit.

**Pull request merged**